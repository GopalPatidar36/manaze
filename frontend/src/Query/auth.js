import { gql } from "@apollo/client";

export const LOGIN = gql`
  query Login($userEmail: String!, $password: String!) {
    login(userEmail: $userEmail, password: $password) {
      token
      user {
        uid
        userEmail
        password
        firstName
        lastName
      }
    }
  }
`;
