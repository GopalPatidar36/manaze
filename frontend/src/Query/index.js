export * from "./tickets";
export * from "./user";