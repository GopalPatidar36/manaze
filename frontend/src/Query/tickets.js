import { gql } from "@apollo/client";

const CORE_FIELDS = gql`
  fragment CoreFields on Ticket {
    id
    title
    description
    priority
    status
  }
`;

const TicketList = gql`
  ${CORE_FIELDS}
  fragment ticketList on TicketList {
    count
    rows {
      ...CoreFields
    }
  }
`;

export const GET_CURRENT_USER_TICKET = gql`
  ${TicketList}
  query CurrentUserTicket($title: String, $description: String, $priority: String, $status: String, $limit: Int, $offset: Int, $order: String) {
    currentUserTicket(title: $title, description: $description, priority: $priority, status: $status, limit: $limit, offset: $offset, order: $order) {
      ...ticketList
    }
  }
`;

export const GET_ALL_TICKET = gql`
  query TicketList($title: String, $description: String, $priority: String, $status: String, $limit: Int, $offset: Int, $order: String) {
    ticketList(title: $title, description: $description, priority: $priority, status: $status, limit: $limit, offset: $offset, order: $order) {
      ...ticketList
    }
  }
  ${TicketList}
`;

export const UPDATE_TICKET = gql`
  mutation UpdateTicket($id: Int!, $title: String, $description: String, $priority: String, $status: String) {
    updateTicket(id: $id, title: $title, description: $description, priority: $priority, status: $status) {
      id
    }
  }
`;
