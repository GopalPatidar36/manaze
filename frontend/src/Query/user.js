import { gql } from "@apollo/client";

const CORE_FIELDS = gql`
  fragment CoreFields on userFields {
    uid
    firstName
    lastName
    role
    createdAt
    updatedAt
    userEmail
  }
`;

const TicketList = gql`
  ${CORE_FIELDS}
  fragment userList on UserSearchResult {
    count
    rows {
      ...CoreFields
    }
  }
`;

export const GET_CURRENT_USER = gql`
  ${CORE_FIELDS}
  query Me {
    me {
      ...CoreFields
    }
  }
`;

export const UPDATE_USER = gql`
  mutation UpdateUser($uid: String!, $firstName: String, $lastName: String) {
    updateUser(uid: $uid, firstName: $firstName, lastName: $lastName) {
      id
    }
  }
`;
