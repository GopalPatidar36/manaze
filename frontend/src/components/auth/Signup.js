import React, { useState } from "react";
import { connect, useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { registerUser } from "../../redux/slices/authSlice";

const Signup = (props) => {
  const dispatch = useDispatch();
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [userEmail, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [confirmPass, setConfirmPass] = useState("");
  const [passError, setPassError] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();
    if (confirmPass !== password) {
      setPassError("Password and Confirm Password do not match. Please try again");
      return;
    }
    dispatch(registerUser({ firstName, lastName, userEmail: userEmail.toLowerCase(), password }));
    setFirstName("");
    setLastName("");
    setEmail("");
    setPassword("");
    setConfirmPass("");
  };

  return (
    <div className={"mainContainer"}>
      <div className="boxContainer">
        <h3>Signup</h3>
        <form className="loginForm" onSubmit={handleSubmit}>
          <input required value={firstName} placeholder="Enter your first name here" onChange={(ev) => setFirstName(ev.target.value)} className={"inputBox"} />

          <input required value={lastName} placeholder="Enter your last name here" onChange={(ev) => setLastName(ev.target.value)} className={"inputBox"} />

          <input
            type="email"
            required
            value={userEmail}
            placeholder="Enter your email here"
            onChange={(ev) => setEmail(ev.target.value)}
            className={"inputBox"}
          />
          <input
            required
            type="password"
            minLength={5}
            value={password}
            placeholder="Enter your password here"
            onChange={(ev) => {
              setPassword(ev.target.value);
              setPassError("");
            }}
            className={"inputBox"}
          />
          <label className="errorLabel">{passError}</label>

          <input
            required
            type="password"
            minLength={5}
            value={confirmPass}
            placeholder="Enter confirm password here"
            onChange={(ev) => {
              setConfirmPass(ev.target.value);
              setPassError("");
            }}
            className={"inputBox"}
          />
          <label className="errorLabel">{passError}</label>

          <button className={"inputButton"} type="submit">
            Signup
          </button>
        </form>
        <Link to="/">Have an account? Log In</Link>
      </div>
    </div>
  );
};

export default connect()(Signup);
