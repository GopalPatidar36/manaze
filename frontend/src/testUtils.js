// src/testUtils.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { Provider } from "react-redux";
import { MockedProvider } from "@apollo/client/testing";
import { BrowserRouter as Router } from "react-router-dom";
import store from "./store";

function render(ui, { preloadedState, apolloMocks, ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <MockedProvider mocks={apolloMocks} addTypename={false}>
          <Router>{children}</Router>
        </MockedProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
