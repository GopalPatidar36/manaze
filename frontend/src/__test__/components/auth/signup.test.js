import React from "react";
import { screen, waitFor, fireEvent } from "@testing-library/react";
import { render } from "../../../testUtils";
import Signup from "../../../components/auth/Signup";
import { CREATE_USER } from "../../../Query/index";

const login = {
  request: {
    query: CREATE_USER,
    variables: { userEmail: "ramp@gmail.com", firstName: "ram", lastName: "patidar", password: "12345" },
  },
  result: { data: { updateUser: { id: 123 } } },
};
const mocks = [login];
describe("Signup page testing", () => {
  let userSetting;
  beforeEach(async () => {
    userSetting = render(<Signup />, { apolloMocks: mocks });
  });

  test("displays personal information header", async () => {
    await waitFor(() => {
      const elements = screen.getAllByText(/Signup/);
      expect(elements[0]).toBeInTheDocument();
    });
  });

  test("enter email and password", async () => {
    await waitFor(() => {
      const firstName = screen.getByPlaceholderText("Enter your first name here");
      fireEvent.change(firstName, { target: { value: "ram" } });

      const lastName = screen.getByPlaceholderText("Enter your last name here");
      fireEvent.change(lastName, { target: { value: "patidar" } });

      const email = screen.getByPlaceholderText("Enter your email here");
      fireEvent.change(email, { target: { value: "ramp@gmail.com" } });

      const password = screen.getByPlaceholderText("Enter your password here");
      fireEvent.change(password, { target: { value: "12345" } });

      const confirm = screen.getByPlaceholderText("Enter confirm password here");
      fireEvent.change(confirm, { target: { value: "123456" } });

      const button = screen.getByRole("button", { name: /Signup/i });
      fireEvent.click(button);
    });

    await waitFor(() => {
      const errorMessage = screen.getAllByText("Password and Confirm Password do not match. Please try again");
      expect(errorMessage[0]).toBeInTheDocument();

      const confirm = screen.getByPlaceholderText("Enter confirm password here");
      fireEvent.change(confirm, { target: { value: "12345" } });
    });

    await waitFor(() => {
      const errorMessage = screen.queryByText("Password and Confirm Password do not match. Please try again");
      expect(errorMessage).not.toBeInTheDocument();
    });
  });

  it("should match dom Login", () => {
    expect(userSetting.asFragment()).toMatchSnapshot();
  });
});
