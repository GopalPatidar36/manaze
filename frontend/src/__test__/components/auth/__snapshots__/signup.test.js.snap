// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup page testing should match dom Login 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <div
      class="boxContainer"
    >
      <h3>
        Signup
      </h3>
      <form
        class="loginForm"
      >
        <input
          class="inputBox"
          placeholder="Enter your first name here"
          required=""
          value=""
        />
        <input
          class="inputBox"
          placeholder="Enter your last name here"
          required=""
          value=""
        />
        <input
          class="inputBox"
          placeholder="Enter your email here"
          required=""
          type="email"
          value=""
        />
        <input
          class="inputBox"
          minlength="5"
          placeholder="Enter your password here"
          required=""
          type="password"
          value=""
        />
        <label
          class="errorLabel"
        />
        <input
          class="inputBox"
          minlength="5"
          placeholder="Enter confirm password here"
          required=""
          type="password"
          value=""
        />
        <label
          class="errorLabel"
        />
        <button
          class="inputButton"
          type="submit"
        >
          Signup
        </button>
      </form>
      <a
        href="/"
      >
        Have an account? Log In
      </a>
    </div>
  </div>
</DocumentFragment>
`;
